### ISSUE TITLE
BUG: fbt is not preserving `&nbsp;`

### ISSUE THREAD

**Issue by mrtnzlml** [facebook/fbt#240]
## 🐛 Bug Report

[Non-breaking space](https://en.wikipedia.org/wiki/Non-breaking_space) `&nbsp;` is being replaced by FBT with simple space. That's unexpected in situations where you really need a non-breaking space.

## To Reproduce

1. create the following test:

```html
<div>
  test&nbsp;1
  <fbt desc="test2">test&nbsp;2</fbt>
</div>
```

2. check the output in your browser, in my case it's this:

```html
<div>test&nbsp;1<!-- -->test 2</div>
```

As you can see, the second `&nbsp;` disappeared which is not correct.

## Expected behavior

`&nbsp;` should be preserved in the output

## envinfo

```bash
  System:
    OS: macOS 11.2.3
    CPU: (8) x64 Intel(R) Core(TM) i5-1038NG7 CPU @ 2.00GHz
    Memory: 637.48 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.3.0 - /usr/local/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 7.15.1 - /usr/local/bin/npm
    Watchman: 20210328.072220.0 - /usr/local/bin/watchman
```

**Comment by mrtnzlml** [comment]
BTW, I tried [`preserveWhitespace`](https://facebook.github.io/fbt/docs/api_intro/#optional-attributes) but that also didn't preserve the `&nbsp;`. 🤔

**Comment by kayhadrin** [comment]
Hi @mrtnzlml, 

That's strange because we're specifically supporting this kind of use-case.

Test case:
https://github.com/facebook/fbt/blob/76debd6aa2c7076a2018ba7ab20a8caa5e9e5ebb/packages/babel-plugin-fbt/src/__tests__/fbtJsx-test.js#L303-L307

Snapshot:
https://github.com/facebook/fbt/blob/6d2de3cbe82f488c0d1180b1fdddea2f8d1e59c1/packages/babel-plugin-fbt/src/__tests__/__snapshots__/fbtJsx-test.js.snap#L543-L553

AFAIK, it's the Babel parser that reads and decodes HTML entities within JSX.
My only guess is that some other Babel transforms before/after the babel-plugin-fbt might have altered this.
Also, how did you verify that the HTML entities were missing in your rendered fbt on the browser?

It'd really help if you could make a repro repository to help us debug this further if necessary. :-)

**Comment by mrtnzlml** [comment]
Hi @kayhadrin, thanks for having a look!

I've created a demo here: https://github.com/mrtnzlml/bug-fbt-nbsp

It is certainly possible that it's something wrong with my setup. However, everything else seems to be working as expected (check the demo with other entities) and I am not sure how should this be fixed in my code. 🤔 

>  Also, how did you verify that the HTML entities were missing in your rendered fbt on the browser?

I've clicked on "Edit as HTML" when inspecting the element because otherwise `&nbsp;` (and some other entities) would be hidden. Output of "Edit as HTML" is visible in the demo README file.

I noticed this issue originally on my other website where "80 ℃" was having an unexpected line-break but it worked as expected when I removed the `<fbt>…</fbt>` wrapper.
